--- python-ethumb-0.7.3/include/ethumb/client.pxd.orig	2010-10-01 01:03:28.000000000 +0200
+++ python-ethumb-0.7.3/include/ethumb/client.pxd	2012-01-22 07:10:02.956174216 +0100
@@ -23,9 +23,11 @@
 
 cdef extern from "Ethumb_Client.h":
     ctypedef struct Ethumb_Client
+    ctypedef struct Ethumb_Exists
     ctypedef void (*Ethumb_Client_Connect_Cb)(void *data, Ethumb_Client *client, Eina_Bool success)
     ctypedef void (*Ethumb_Client_Die_Cb)(void *data, Ethumb_Client *client)
     ctypedef void (*Ethumb_Client_Generate_Cb)(void *data, Ethumb_Client *client, int id, char *file, char *key, char *thumb_path, char *thumb_key, Eina_Bool success)
+    ctypedef void (*Ethumb_Client_Thumb_Exists_Cb)(Ethumb_Client *client, Ethumb_Exists *thread, Eina_Bool exists, void *data)
     ctypedef void (*Ethumb_Client_Generate_Cancel_Cb)(void *data, Eina_Bool success)
     ctypedef void (*Eina_Free_Cb)(void *data)
 
@@ -68,7 +70,9 @@
 
     void ethumb_client_thumb_path_set(Ethumb_Client *client, char *path, char *key)
     void ethumb_client_thumb_path_get(Ethumb_Client *client, char **path, char **key)
-    Eina_Bool ethumb_client_thumb_exists(Ethumb_Client *client)
+    Ethumb_Exists *ethumb_client_thumb_exists(Ethumb_Client *client, Ethumb_Client_Thumb_Exists_Cb exists_cb, void *data)
+    void ethumb_client_thumb_exists_cancel(Ethumb_Exists *exists, Ethumb_Client_Thumb_Exists_Cb exists_cb, void *data)
+    Eina_Bool ethumb_client_thumb_exists_check(Ethumb_Exists *exists)
     int ethumb_client_generate(Ethumb_Client *client, Ethumb_Client_Generate_Cb generated_cb, void *data, Eina_Free_Cb free_data)
     void ethumb_client_generate_cancel(Ethumb_Client *client, int id, Ethumb_Client_Generate_Cancel_Cb cancel_cb, void *data, Eina_Free_Cb free_data)
     void ethumb_client_generate_cancel_all(Ethumb_Client *client)
--- python-ethumb-0.7.3/ethumb/ethumb.client.pyx.orig	2012-01-22 07:25:59.802872920 +0100
+++ python-ethumb-0.7.3/ethumb/ethumb.client.pyx	2012-01-22 07:36:52.716228115 +0100
@@ -544,13 +544,14 @@
     def document_page_set(self, int page):
         ethumb_client_document_page_set(self.obj, page)
 
-    def exists(self):
-        """Checks if thumbnail already exists.
+    # XXX: now it's to be done in async way
+    #def exists(self):
+    #    """Checks if thumbnail already exists.
 
-        If you want to avoid regenerating thumbnails, check if they
-        already exist with this function.
-        """
-        return bool(ethumb_client_thumb_exists(self.obj))
+    #    If you want to avoid regenerating thumbnails, check if they
+    #    already exist with this function.
+    #    """
+    #    return bool(ethumb_client_thumb_exists(self.obj))
 
     def generate(self, func, *args, **kargs):
         """Ask EThumb server to generate the specified thumbnail.
